packages/superset-ui-chart/src/types/Query.ts

```diff
@@ -38,8 +38,8 @@ export interface PlainProps {
 
 type TransformFunction<Input = PlainProps, Output = PlainProps> = (x: Input) => Output;
 
-export type PreTransformProps = TransformFunction<ChartProps>;
-export type TransformProps = TransformFunction;
+export type PreTransformProps = TransformFunction<ChartProps, ChartProps>;
+export type TransformProps = TransformFunction<ChartProps>;
```

Comment:
I tried to do like 
```
export type PreTransformProps = TransformFunction<ChartProps, ChartProps | PlainProps>;
export type TransformProps = TransformFunction<ChartProps | PlainProps>;`
```

However, in Chart plugin if I define `TransfromProps` as 

```
export default function transformProps(chartProps: ChartProps)
```

it will show the error: `Type 'PlainProps | ChartProps' is not assignable to type 'ChartProps'.`